src/main/User/UserController.java

```diff
@@ -271,4 +279,35 @@ public static JSONObject mergeJSON(
     }
     return merged;
   }
+
+  public InputStream getUserPfp(String username) {
+    Bson filter = Filters.eq("metadata.owner", username);
+    GridFSBucket gridBucket = GridFSBuckets.create(db, "pfp");
+    GridFSFile grid_out = gridBucket.find(filter).first();
+    if (grid_out == null || grid_out.getMetadata() == null) {
+      return null;
+    }
+    InputStream pfp = gridBucket.openDownloadStream(grid_out.getObjectId());
+    return pfp;
+  }
+
+  public Handler uploadPfp =
+      ctx -> {
+        String username = ctx.sessionAttribute("username");
+        UploadedFile file = ctx.uploadedFile("file");
+        logger.info(username + " is attempting to upload a profile picture");
+        UploadPfpService serv = new UploadPfpService(db, logger, username, file);
+        JSONObject res = serv.executeAndGetResponse().toJSON();
+        logger.info(username + " has successfully uploaded a profile picture");
+        ctx.json(res.toString());
+      };
+
+  public Handler loadPfp =
+      ctx -> {
+        String username = ctx.sessionAttribute("username");
+        LoadPfpService lps = new LoadPfpService(db, logger, username);
+        JSONObject mes = lps.executeAndGetResponse().toJSON();
+        ctx.header("Content-Type", "application/pfp");
```

Comment:
also i dont think this is a correct content-type header. see: https://www.geeksforgeeks.org/http-headers-content-type/